src/sinks/elasticsearch/retry.rs

```diff
@@ -127,21 +153,58 @@ impl RetryLogic for ElasticsearchRetryLogic {
                                 // We will retry if there exists at least one item that
                                 // failed with a retriable error.
                                 // Those are backpressure and server errors.
-                                if let Some((status, error)) =
+                                let status_codes: Vec<bool> = resp
+                                    .iter_status()
+                                    .map(|(status, _)| {
+                                        status == StatusCode::TOO_MANY_REQUESTS
+                                            || status.is_server_error()
+                                    })
+                                    .collect();
```

Comment:
Nit: The second iteration can be avoided by doing the following:

```suggestion
							    let retriable = status == StatusCode::TOO_MANY_REQUESTS || status.is_server_error();
							    retry_flags.push(retriable);
							    if retriable {
							        should_retry_partial = true;
							    }
							}
```

Reply:
Bump on this